dom.bs

```diff
@@ -6185,54 +6185,54 @@ given a <var>document</var>, <var>localName</var>, <var>namespace</var>, and opt
 <a>attribute</a> <var>attribute</var> to <var>value</var>, run these steps:
 
 <ol>
+ <li><p>Set <var>attribute</var>'s <a for=Attr>value</a> to <var>value</var>.
+
  <li><p><a>Handle attribute changes</a> for <var>attribute</var> with <var>attribute</var>'s
  <a for=Attr>element</a>, <var>attribute</var>'s <a for=Attr>value</a>, and <var>value</var>.
-
- <li><p>Set <var>attribute</var>'s <a for=Attr>value</a> to <var>value</var>.
 </ol>
 
 <p>To <dfn export id=concept-element-attributes-append lt="append an attribute">append</dfn> an
 <a>attribute</a> <var>attribute</var> to an <a for=/>element</a> <var>element</var>, run these
 steps:
 
 <ol>
- <li><p><a>Handle attribute changes</a> for <var>attribute</var> with <var>element</var>, null, and
- <var>attribute</var>'s <a for=Attr>value</a>.
-
  <li><p><a for=list>Append</a> <var>attribute</var> to <var>element</var>'s
  <a for=Element>attribute list</a>.
 
  <li><p>Set <var>attribute</var>'s <a for=Attr>element</a> to <var>element</var>.
+
+ <li><p><a>Handle attribute changes</a> for <var>attribute</var> with <var>element</var>, null, and
+ <var>attribute</var>'s <a for=Attr>value</a>.
 </ol>
 
 <p>To <dfn export id=concept-element-attributes-remove lt="remove an attribute">remove</dfn> an
 <a>attribute</a> <var>attribute</var>, run these steps:
 
 <ol>
- <li><p><a>Handle attribute changes</a> for <var>attribute</var> with <var>attribute</var>'s
- <a for=Attr>element</a>, <var>attribute</var>'s <a for=Attr>value</a>, and null.
-
  <li><a for=list>Remove</a> <var>attribute</var> from <var>attribute</var>'s
  <a for=Attr>element</a>'s <a for=Element>attribute list</a>.
 
  <li><p>Set <var>attribute</var>'s <a for=Attr>element</a> to null.
+
+ <li><p><a>Handle attribute changes</a> for <var>attribute</var> with <var>attribute</var>'s
+ <a for=Attr>element</a>, <var>attribute</var>'s <a for=Attr>value</a>, and null.
```

Comment:
Presumably we need to cache attribute's element beforehand?

Also, it's surprising that for element removal it runs before, but for attribute removal it's after.

Did you double check what implementations do here?

Reply:
> Presumably we need to cache attribute's element beforehand?

Done

> Also, it's surprising that for element removal it runs before, but for attribute removal it's after.
> 
> Did you double check what implementations do here?

There's things that run before and after element removal in chromium: https://github.com/whatwg/dom/pull/1185#issuecomment-1520926675

As for attribute changes, I just did a quick look and it does not look like there is anything which runs prior to an attribute change in chromium.